### PR TITLE
test: E2E framework

### PR DESCRIPTION
**Pull request by matmerr**
**Reason for Change**:

Add a programmatic way for deploying e2e resources in lieu of bash/make/ps scripts

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] relevant PR labels added

**Notes**:

### CODE REVIEW

File: .pipelines/pipeline.yaml [hunk 1]
```diff
@@ -53,555 +53,555 @@ stages:
             displayName: "Set environmental variables"
             condition: always()
   - ${{ if contains(variables['Build.SourceBranch'], 'refs/pull') }}:
```
Comment (jpayne3506) [inline .pipelines/pipeline.yaml:55]: Is the only change a tab difference? Hard to see from the diff

Comment (matmerr) [reply .pipelines/pipeline.yaml]: yeah tab diff from formatter, reverting

File: test/e2e/manifests/cilium/v1.14/agent/cilium-config-hubble.yaml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  enable-metrics: "true"
+  enable-policy: default
+  enable-remote-node-identity: "true"
+  enable-session-affinity: "true"
+  enable-svc-source-range-check: "true"
+  enable-vtep: "false"
+  enable-well-known-identities: "false"
+  enable-xt-socket-fallback: "true"
+  hubble-metrics-server: :9965
+  hubble-metrics: flow:sourceEgressContext=workload-name;destinationIngressContext=workload-name tcp:sourceEgressContext=workload-name;destinationIngressContext=workload-name dns:query;sourceEgressContext=workload-name;destinationIngressContext=workload-name drop:sourceEgressContext=workload-name;destinationIngressContext=workload-name
```
Comment (matmerr) [inline test/e2e/manifests/cilium/v1.14/agent/cilium-config-hubble.yaml:46]: remove dns

File: test/e2e/manifests/cilium/v1.14/agent/cilium-config-hubble.yaml [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  enable-health-checking: "true"
+  enable-host-legacy-routing: "true"
+  enable-hubble: "true"
+  enable-ipv4: "true"
+  enable-ipv4-masquerade: "false"
+  enable-ipv6: "false"
+  enable-ipv6-masquerade: "false"
+  enable-k8s-terminating-endpoint: "true"
+  enable-l2-neigh-discovery: "true"
+  enable-l7-proxy: "true"
```
Comment (matmerr) [inline test/e2e/manifests/cilium/v1.14/agent/cilium-config-hubble.yaml:34]: set false

**Pull request closed without merging**